src/server.ts

```diff
@@ -75,7 +75,7 @@ export class SubscriptionServer {
         // then terminates the actual network connection (sends FIN packet)
         // 1002: protocol error
         request.close(1002);
-        request.terminate();
+        setTimeout(() => request.terminate(), 10);
```

Comment:
I was seeing race conditions in the tests such that the `close` was being superseded by the `terminate`.